install/resources/upgrade34.php

```diff
@@ -0,0 +1,46 @@
+<?php
+/**
+ * MyBB 1.8
+ * Copyright 2014 MyBB Group, All Rights Reserved
+ *
+ * Website: http://www.mybb.com
+ * License: http://www.mybb.com/about/license
+ *
+ */
+
+/**
+ * Upgrade Script: 1.8.5
+ */
+
+$upgrade_detail = array(
+	"revert_all_templates" => 0,
+	"revert_all_themes" => 0,
+	"revert_all_settings" => 0
+);
+
+@set_time_limit(0);
+
+function upgrade34_dbchanges()
+{
+	global $db, $output;
+
+	$query = $db->simple_select('settings', 'value', "name='username_method'");
```

Comment:
> The part of code you reference only takes place if the cache file does not exists.

Well, if we really want to ensure the settings match database in every case during the upgrade, not just randomly in one update script, the file cache check should be removed and one global setting query should be ran instead of wasting one query per every setting. Otherwise it means the cache is trustable enough to not do that.